backend/inspirehep/search/facets.py

```diff
@@ -30,8 +30,22 @@ def inner(values):
     return inner
 
 
+def must_match_all_filter_nested(nested_path, match_field):
+    """Bool filter containing a list of must matches for nested queries."""
+
+    def inner(values):
+        filters = []
+        for value in values:
+            filters.append(Q("match", **{match_field: value}))
```

Comment:
The common pattern
```python
out_list = []
for element in in_list:
    if p(element):
        out_list.append(f(element))
```
can be expressed more clearly and concisely as a list comprehension
```python
out_list = [f(element) for element in in_list if p(element)]
```